src/sdk/pynni/nni/evolution_tuner/evolution_tuner.py

```diff
@@ -168,17 +207,41 @@ def generate_parameters(self, parameter_id, **kwargs):
                 is_rand[self.space[i]] = (self.space[i] == mutation_pos)
             config = json2parameter(
                 self.searchspace_json, is_rand, self.random_state, self.population[0].config)
-            self.population.pop(1)
-            # remove "_index" from config and save params-id
 
-            total_config = config
+            if not self.population:
+                self.population.pop(1)
```

Comment:
what if `self.population` only has one element?

Reply:
Thanks~ I have fixed it, please review on the latest version.